### ISSUE TITLE
squinky labs plugin won't load from latest release (libpthread)

### ISSUE THREAD

**Issue by squinkylabs** [VCVRack/community#475]
The new build from plugin manager is not loading the plugin for some people. We are seeing the same issue.

Using the downloaded version of VCV Rack for Windows (0.6.0) we get the "can't load plugin" dialog, and the log shows error 127. If I copy over a plugin.dll built locally, it works.

One odd thing is that the sizes of the dll files are quite different. When we build it locally it's about 5 megabytes, but the downloaded one is 463k bytes.

We do our builds against v0.6.0 branch in rack.

**Comment by squinkylabs** [comment]
Have verified that this does not affect mac. The mac version works fine from download manager.

**Comment by squinkylabs** [comment]
I don't know what to make of the size issue. When I build it is about 5M, but make disk creates one that is the more sane "reduced" size for 453k. When I unzip the stuff in the zip that `make dist` made it works. I even posted that as a release, and I can download it and unzip it and it works.

**Comment by martin-lueders** [comment]
Permissions look fine to me, but I find that the binary (*.dll) in my dist directory is 454KB, while the one unpacked from the plugin manager is 453KB. Might be a blocking thing, though...

**Comment by squinkylabs** [comment]
Mine tend to be more around 463k. Probably depends on exactly which versions of gcc and the other tools we are using, so hard to know what to make of it.

**Comment by squinkylabs** [comment]
v0.6.0. Sometimes I dabble with current master in a different repo, but for most work 0.6.0

**Comment by dizzisound** [comment]
@squinkylabs The released version on this repo works fine for me, too (Windows 7 x64). The version downloaded via Plugin Manager doesn't load (so, I got the same issue raised here).
Same sizes, too, but I found that the _dll_ fetched from the PluginManager is missing the link to `libwinpthread`.
Attaching a couple of screenshot, with the scans of the two versions.

_PM version_
![squinky-0 6 4-pm](https://user-images.githubusercontent.com/36040812/40631676-3bb4ca3c-62df-11e8-8bdc-d2e03cf75f5e.jpg)

_This repo version_
![squinky-0 6 4-repo](https://user-images.githubusercontent.com/36040812/40631679-3e056184-62df-11e8-9ba3-58f875b5532e.jpg)

**Comment by squinkylabs** [comment]
@dizzisound - looks like you found it!!! Now that you mention it, I never though to wonder how the linker finds libpthread. I guess because it links in librackRack.a, which presumably does. maybe the build machines are using a version or rack that uses static linking and hence doesn't pull that in for free. Maybe I need to add -lpthread to my linker? An any case, will need some VCV linker experts to weigh in on how to fix this.

**Comment by AndrewBelt** [comment]
Thanks @dizzisound, that solved it.
Since the mostly-automated build process uses the Rack-SDK of recent versions of the `v0.6` branch instead of the most recent official release (because some plugins use parts of the bleeding-edge API which is ABI-compatible with Rack 0.6.0), and since Rack has switched to a full static build on Windows (so relieved I finally got around to that), there's no implicit linking to mingw64's pthreads from plugins which use that library.

The solution is to add `-lpthread` to
https://github.com/squinkylabs/SquinkyVCV/blob/master/Makefile#L30
before you send your next update.
I've already corrected this for you in this version by modifying my local copy of your source. Since the plugin currently doesn't load for Windows users, an update to the plugin will be triggered when they hit Update Plugins, and it will download the corrected version.
